src/ChunkMap.cpp

```diff
@@ -2718,41 +2719,83 @@ void cChunkMap::TickBlock(int a_BlockX, int a_BlockY, int a_BlockZ)
 
 
 
-void cChunkMap::UnloadUnusedChunks(void)
+size_t cChunkMap::UnloadUnusedChunks(size_t a_Amount)
 {
+	size_t UnloadedTotal = 0;
+	bool UnloadAll;
+
+	if (a_Amount == 0)
+	{
+		UnloadAll = true;
+		a_Amount = m_World->GetNumChunks();
+	}
+	else
+	{
+		UnloadAll = false;
+	}
+
+	if (a_Amount == 0)
+	{
+		return 0;
+	}
+
 	cCSLock Lock(m_CSLayers);
-	for (cChunkLayerList::iterator itr = m_Layers.begin(); itr != m_Layers.end(); ++itr)
+	if (UnloadAll)
 	{
-		(*itr)->UnloadUnusedChunks();
-	}  // for itr - m_Layers
+		for (auto Layer : m_Layers)
+		{
+			Layer->UnloadUnusedChunks(a_Amount);
+		}
+		return a_Amount;
+	}
+	else
+	{
+		cFastRandom Rand;
+		int Size = static_cast<int>(m_Layers.size());
+		// Choose 10 layers randomly and unload them.
+		// Stop if we've unloaded enough.
+		for (int i = 0; i < 10; ++i)
+		{
+			int ChosenLayerNum = Rand.NextInt(Size);
```

Comment:
I can't actually find a code-path that frees cChunkLayer objects if before chunk destruction. So Size can only increase, and this is safe. But we should probably look at freeing them.